diff -urN obexftp-0.23/swig/ruby/ruby_wrap.c obexftp-0.23-patched/swig/ruby/ruby_wrap.c
--- obexftp-0.23/swig/ruby/ruby_wrap.c	2009-02-18 04:30:40.000000000 +1000
+++ obexftp-0.23-patched/swig/ruby/ruby_wrap.c	2012-09-18 02:40:01.595106852 +1100
@@ -1309,7 +1309,7 @@
 /* Error manipulation */
 
 #define SWIG_ErrorType(code)                            SWIG_Ruby_ErrorType(code)               
-#define SWIG_Error(code, msg)            		rb_raise(SWIG_Ruby_ErrorType(code), msg)
+#define SWIG_Error(code, msg)            		rb_raise(SWIG_Ruby_ErrorType(code), "%s", msg)
 #define SWIG_fail                        		goto fail				 
 
 
