diff -Naur obexftp-0.23.orig/swig/charmap.i obexftp-0.23/swig/charmap.i
--- obexftp-0.23.orig/swig/charmap.i	2012-09-12 19:36:12.000000000 +0400
+++ obexftp-0.23/swig/charmap.i	2012-09-12 19:36:57.000000000 +0400
@@ -88,9 +88,8 @@
 	$1 = PyString_AsString($input);
 	$2 = PyString_Size($input);
 #elif defined SWIGRUBY
-/* VALUE str = StringValue($input); // perhaps better? */
-	$1 = STR2CSTR($input);
-	$2 = (int) RSTRING($input)->len;
+	$1 = RSTRING_PTR($input);
+	$2 = (int) RSTRING_LEN($input);
 #elif defined SWIGTCL
 	$1 = Tcl_GetStringFromObj($input,&$2);
 #else
diff -Naur obexftp-0.23.orig/swig/ruby/ruby_wrap.c obexftp-0.23/swig/ruby/ruby_wrap.c
--- obexftp-0.23.orig/swig/ruby/ruby_wrap.c	2012-09-12 19:36:12.000000000 +0400
+++ obexftp-0.23/swig/ruby/ruby_wrap.c	2012-09-12 19:36:18.000000000 +0400
@@ -2580,7 +2580,7 @@
     
     /* VALUE str = StringValue(argv[0]); // perhaps better? */
     arg2 = STR2CSTR(argv[0]);
-    arg3 = (int) RSTRING(argv[0])->len;
+    arg3 = (int) RSTRING_LEN(argv[0]);
     
     
     
